src/pandablocks_ioc/_pvi.py

```diff
@@ -207,21 +217,19 @@ def create_pvi_records(record_prefix: str):
         device_refs = [DeviceRef(x, x) for x in pvi_records]
 
         # # TODO: What should the label be?
-        device = Device("TOP", device_refs)
+        device = Device("TOP", children=device_refs)
         devices.append(device)
 
         # TODO: label widths need some tweaking - some are pretty long right now
         formatter = DLSFormatter(label_width=250)
-        with tempfile.TemporaryDirectory() as temp_dir:
-            for device in devices:
-                try:
-                    formatter.format(
-                        device,
-                        record_prefix + ":",
-                        Path(f"{temp_dir}/{device.label}.bob"),
-                    )
-                    with open(f"{temp_dir}/{device.label}.bob") as f:
-                        Pvi.bob_file_dict.update({f"{device.label}.bob": f.read()})
 
-                except NotImplementedError:
-                    logging.exception("Cannot create TABLES yet")
+        for device in devices:
+            bobfile_path = Pvi._screens_dir / Path(f"{device.label}.bob")
+            if not overwrite_bobfiles:
+                if bobfile_path.is_file():
+                    raise FileExistsError(
+                        f"Trying to write bobfile for {device.label}, but File "
+                        f"{bobfile_path} already exists. Use --overwrite-bobfiles "
+                        "to enable overwrite."
+                    )
+            formatter.format(device, record_prefix + ":", bobfile_path)
```

Comment:
In the case of `overwrite_bobfiles == False`, this code will have the odd effect of creating new files, and then stopping when the first already existing one is present. Is this the behaviour we want, or would it be better to check ahead of time that we aren't going to attempt to override anything before writing?

Reply:
Maybe best to check ahead of time, I'll change it.

Reply:
I wrote some stuff for this, but after discussions with Tom we think it would be safer to drop the new argument and just make sure the screens directory is completely empty before doing any writing.